// culqi-node/charges.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.charges = void 0;
const request_1 = require("./request");
const vars_1 = __importDefault(require("./vars"));
exports.charges = {
    createCharge: (req, extraHttpOptions) => (0, request_1.post)(vars_1.default.basePaths.charges, req, extraHttpOptions),
    getCharge: (req, extraHttpOptions) => (0, request_1.get)(`${vars_1.default.basePaths.charges}/${req.id}`, undefined, extraHttpOptions),
    getCharges: (req, extraHttpOptions) => (0, request_1.get)(vars_1.default.basePaths.charges, req, extraHttpOptions),
    updateCharge: (req, extraHttpOptions) => (0, request_1.patch)(`${vars_1.default.basePaths.charges}/${req.id}`, req, extraHttpOptions),
    captureCharge: (req, extraHttpOptions) => (0, request_1.post)(`${vars_1.default.basePaths.charges}/${req.id}/capture`, req, extraHttpOptions),
};
