// culqi-node/customers.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.customers = void 0;
const request_1 = require("./request");
const vars_1 = __importDefault(require("./vars"));
exports.customers = {
    createCustomer: (req, extraHttpOptions) => (0, request_1.post)(vars_1.default.basePaths.customers, req, extraHttpOptions),
    getCustomer: (req, extraHttpOptions) => (0, request_1.get)(`${vars_1.default.basePaths.customers}/${req.id}`, undefined, extraHttpOptions),
    getCustomers: (req, extraHttpOptions) => (0, request_1.get)(vars_1.default.basePaths.customers, req, extraHttpOptions),
    updateCustomer: (req, extraHttpOptions) => (0, request_1.patch)(`${vars_1.default.basePaths.customers}/${req.id}`, req, extraHttpOptions),
    deleteCustomer: (req, extraHttpOptions) => (0, request_1.delete_)(`${vars_1.default.basePaths.customers}/${req.id}`, extraHttpOptions),
};
