// culqi-node/orders.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.orders = void 0;
const request_1 = require("./request");
const vars_1 = __importDefault(require("./vars"));
exports.orders = {
    createOrder: (req, extraHttpOptions) => (0, request_1.post)(vars_1.default.basePaths.orders, req, extraHttpOptions),
    confirmOrder: (req, extraHttpOptions) => (0, request_1.post)(`${vars_1.default.basePaths.orders}/${req.id}/confirm`, req, extraHttpOptions),
    getOrder: (req, extraHttpOptions) => (0, request_1.get)(`${vars_1.default.basePaths.orders}/${req.id}`, undefined, extraHttpOptions),
    getOrders: (req, extraHttpOptions) => (0, request_1.get)(vars_1.default.basePaths.orders, req, extraHttpOptions),
    updateOrder: (req, extraHttpOptions) => (0, request_1.patch)(`${vars_1.default.basePaths.orders}/${req.id}`, req, extraHttpOptions),
    deleteOrder: (req, extraHttpOptions) => (0, request_1.delete_)(`${vars_1.default.basePaths.orders}/${req.id}`, extraHttpOptions),
};
