// culqi-node/subscriptions.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriptions = void 0;
const request_1 = require("./request");
const vars_1 = __importDefault(require("./vars"));
exports.subscriptions = {
    createSubscription: (req, extraHttpOptions) => (0, request_1.post)(vars_1.default.basePaths.subscriptions, req, extraHttpOptions),
    getSubscription: (req, extraHttpOptions) => (0, request_1.get)(`${vars_1.default.basePaths.subscriptions}/${req.id}`, undefined, extraHttpOptions),
    getSubscriptions: (req, extraHttpOptions) => (0, request_1.get)(vars_1.default.basePaths.subscriptions, req, extraHttpOptions),
    updateSubscription: (req, extraHttpOptions) => (0, request_1.patch)(`${vars_1.default.basePaths.subscriptions}/${req.id}`, req, extraHttpOptions),
    deleteSubscription: (req, extraHttpOptions) => (0, request_1.delete_)(`${vars_1.default.basePaths.subscriptions}/${req.id}`, extraHttpOptions),
};
